src/engine/pieces/piece.js

```diff
@@ -24,6 +24,25 @@ export default class Piece {
         return inside
     }
 
+    createMovesArray(board, possibleDir) {
+        let currentSquare = board.findPiece(this);
+        let availableMoves = []
+        possibleDir.forEach(dir => {
+            let row = currentSquare.row + dir[0];
+            let col = currentSquare.col + dir[1];
+            if (this.inBoundary(row, col)) {
```

Comment:
I think we can 'invert' this if statement to reduce nesting. ie we could do something like:
```
if (!this.inBoundary(row, col) {
    return;
}
// rest of code
```